Add shell detection support and exit if cargo not installed

devanoneth opened the pull request:
I personally use zsh instead of bash. This meant I got an error in L64 of the original `install` script as zsh uses a `.zshrc` file instead of a `.profile` file in $HOME.

This PR:
- adds initial shell detection (might be helpful in future)
- supports `.zshrc` files
- and also, because it affected me too, checks if `cargo` is installed before `forgeup` tries to run

Now my forge journey continues. :)

devanoneth committed:
Add switch for detecting shell and handle ZSH

devanoneth committed:
Clean up colors used and add note regarding cargo if not installed

devanoneth committed:
Return exit code of 1 for error, change e.g. to i.e.

transmissions11 commented on install:
```diff
@@ -60,14 +66,29 @@ mkdir -p $FORGE_DIR
-source $HOME/.profile
\ No newline at end of file
+# Source the updated profile if we are in bash
+if [[ "$SHELL" == *"bash"* ]]; then
```
can you add an else statement for sourcing zshrc

devanoneth replied:
I tried that actually, but the script runs in bash because it starts with `#!/usr/bin/env bash`. So, if you try to source the `.zshrc` file then, it actually throws an error.  We obviously shouldn't change it to run in zsh as it's not as ubiquitous.

I'm not sure if there is a better way to reload the `.zshrc` from within a script which needs to work in both bash and zsh. From what I've seen, telling the user to manually reload the shell is quite common, and I guess that's why.

devanoneth replied:
FWIW, I just installed rust and even that tells you to reload your shell rather than trying to do it automatically.

<img width="494" alt="Screenshot 2022-01-15 at 19 49 34" src="https://user-images.githubusercontent.com/89424366/149634248-a6add74a-4206-4d2b-b912-894e7a2f426f.png">

transmissions11 replied:
alright we should do that unilaterally as well then for consistency imo

devanoneth replied:
Cool, agreed. Pushed a commit which should address that nicely now.

devanoneth committed:
Escape $PATH when exporting it into profile 

Otherwise it prints the PATH variable as it currently is to the profile and could block other changes to PATH in the future

devanoneth committed:
Instruct user to reload PATH rather than sourcing

devanoneth committed:
Remove source instruction from README

transmissions11 commented:
good stuff, thanks king
